### PR TITLE
External tools fix

### LINKED ISSUE #9453: External tool problems when using allowedApiCalls / signed urls
**What steps does it take to reproduce the issue?**

1. I'm running Dataverse 5.13 form the master branch
2. I tried following the External tools documentation https://guides.dataverse.org/en/latest/api/external-tools.html and load the [dynamicDatasetTool.json](https://guides.dataverse.org/en/latest/_downloads/11454dc75acef5448a10509ac962260d/dynamicDatasetTool.json) example
3. After that, trying to access the "Dynamic Dataset Tool" it won't get he "callback" param with the signed URL-s. This is because the in the documentation the "allowedApiCalls" key is under "toolParameters", however in [ExternalToolServiceBean.java ](https://github.com/IQSS/dataverse/blob/04a56e25435f465de4ee80ae540b7596d08b7a24/src/main/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBean.java#L204) it is parsed from the manifest's root (from jsonObject instead of toolParametersObj)
4. After fixing the manifest json and moving "allowedApiCalls" to the root of the json when I try to access the "Dynamic Dataset Tool" I get NPE:

```
  StandardWrapperValve[Faces Servlet]: Servlet.service() for servlet Faces Servlet threw exception
java.lang.NullPointerException
	at edu.harvard.iq.dataverse.externaltools.ExternalToolHandler.handleRequest(ExternalToolHandler.java:116)
	at edu.harvard.iq.dataverse.externaltools.ExternalToolHandler.handleRequest(ExternalToolHandler.java:83)
	at edu.harvard.iq.dataverse.externaltools.ExternalToolHandler.getToolUrlWithQueryParams(ExternalToolHandler.java:229)
	at edu.harvard.iq.dataverse.externaltools.ExternalToolHandler.getExploreScript(ExternalToolHandler.java:251)
	at edu.harvard.iq.dataverse.DatasetPage.explore(DatasetPage.java:5734)
	...
]]
```

In [ExternalToolHandler](https://github.com/IQSS/dataverse/blob/04a56e25435f465de4ee80ae540b7596d08b7a24/src/main/java/edu/harvard/iq/dataverse/externaltools/ExternalToolHandler.java#L110-L118) there seems to be a "break" missing in `case DATASET:`, so if I defined an external tool with `"scope": "dataset"`, it will also reach the `case FILE:` and will cause NPE for `dataFile.getId()`

**What did you expect to happen?**

The documentation suggests that the API Token way is `deprecated, less secure, not recommended` and so I thought the new signed url method should work.

It seems the unit tests don't cover either of the problems above.

**Which version of Dataverse are you using?**

5.13 from master branch.

### PR DESCRIPTION
**Pull request by beepsoft**
**What this PR does / why we need it**:

Fixes bugs with exernal tools.

**Which issue(s) this PR closes**:

- closes #9453

### CODE REVIEW

**Comment by pdurbin** [comment]
downloadAllFilesByVersion failed: https://jenkins.dataverse.org/job/IQSS-Dataverse-Develop-PR/job/PR-9454/1/testReport/edu.harvard.iq.dataverse.api/DownloadFilesIT/downloadAllFilesByVersion/

Screenshot 2023-03-17 at 14-13-13 IQSS-Dataverse-Develop-PR_PR-9454 #1 test - downloadAllFilesByVersion Jenkins

**Comment by qqmyers** [comment]
Pretty hard to see how that test could be related. Is it one that is known to be timing sensitive?

**Comment by beepsoft** [comment]
If you mean the one in [ExternalToolHandler.java](https://github.com/IQSS/dataverse/pull/9454/commits/2805d8aa07fbfa75ebaea2fca25942ffadf75639#diff-b230b5b23faf4f56423f89911a9d7cc145a505b01f47d1db3b2720ed87297a2e), then yes, that is the fix for the NPE.

**Comment by qqmyers** [comment]
FWIW: It looks like we have something else causing the test failure - we'll try to rerun this PR once that's understood/resolved.

**Comment by beepsoft** [comment]
I tried

```
mvn test -P all-unit-tests
```

And this was my result:

```
[WARNING] Tests run: 1510, Failures: 0, Errors: 0, Skipped: 8
```

**Comment by pdurbin** [comment]
@beepsoft it was an integration test that failed (probably nothing to do with this PR). Running them is more involved, I'm afraid: https://guides.dataverse.org/en/5.13/developers/testing.html#integration-tests

File: src/main/java/edu/harvard/iq/dataverse/externaltools/ExternalToolHandler.java [hunk 1]
```diff
@@ -111,6 +111,7 @@ public String handleRequest(boolean preview) {
                 case DATASET:
                     callback=SystemConfig.getDataverseSiteUrlStatic() + "/api/v1/datasets/"
                             + dataset.getId() + "/versions/:latest/toolparams/" + externalTool.getId();
+                    break;
                 case FILE:
                     callback= SystemConfig.getDataverseSiteUrlStatic() + "/api/v1/files/"
                             + dataFile.getId() + "/metadata/" + fileMetadata.getId() + "/toolparams/"
```
Comment (pdurbin) [inline src/main/java/edu/harvard/iq/dataverse/externaltools/ExternalToolHandler.java:117]: I'm just noting that this newer code (inside the `else`) doesn't have any test coverage.

From https://coveralls.io/builds/60355776/source?filename=src%2Fmain%2Fjava%2Fedu%2Fharvard%2Fiq%2Fdataverse%2Fexternaltools%2FExternalToolHandler.java

Screen Shot 2023-05-26 at 4 46 21 PM

**Review by pdurbin — APPROVED** [review]
I didn't test it but the fix makes sense. Same with the docs.

**Pull request merged**